documentation/modules/proc-manual-restart-connector.adoc

```diff
@@ -0,0 +1,27 @@
+// Module included in the following assemblies:
+//
+//
+
+[id='proc-manual-restart-connector-{context}']
+= Performing a restart of a Kafka connector
+
+This procedure describes how to manually trigger a restart of a Kafka connector by using a Kubernetes annotation.
+
+.Prerequisites
+
+* The Cluster Operator is running.
+
+.Procedure
+
+. Find the name of the `KafkaConnector` custom resource that controls the Kafka connector you want to restart:
+[source,shell,subs=+quotes]
+kubectl get KafkaConnector
+
+. To restart the connector, annotate the `KafkaConnector` resource in Kubernetes.
+For example, using `kubectl annotate`:
+[source,shell,subs=+quotes]
+kubectl annotate KafkaConnector _KafkaConnector-name_ strimzi.io/restart=true
```

Comment:
I wonder if this should be `restart-connector`, so it's consistent with `restart-task`?

Reply:
Similar questions around this were raised in the proposal discussion:
 - https://github.com/strimzi/proposals/pull/8#discussion_r453168363
 - https://github.com/strimzi/proposals/pull/8#discussion_r453168427
 - https://github.com/strimzi/proposals/pull/8#discussion_r457633852
 
 For `KafkaConnnector` this PR has the 2 annotations which apply to the resource they are being applied to:
  - `strimzi.io/restart=true`
  - `strimzi.io/restart-task=<TASK-ID>`
 
 And for `KafkaMirrorMaker2` this PR supports annotations that refer to a single connector or task amongst potentially many that are managed by the resource:
  - `strimzi.io/restart-connector=<MM2-CONNECTOR-NAME>`
  - `strimzi.io/restart-connector-task=<MM2-CONNECTOR-NAME>:<TASK-ID>`
  
With this style it's clear which annotations apply to which resource (and which connector/task is being restarted). 

If people think it would be clearer, I am fine to change this to use the same annotations for both `KafkaConnnector` and `KafkaMirrorMaker2`, - e.g:

For `KafkaConnnector`:
  - `strimzi.io/restart-connector=true`
  - `strimzi.io/restart-connector-task=<TASK-ID>`
  
For `KafkaMirrorMaker2`:
  - `strimzi.io/restart-connector=<MM2-CONNECTOR-NAME>`
  - `strimzi.io/restart-connector-task=<MM2-CONNECTOR-NAME>:<TASK-ID>`

What do you think?

Reply:
I _think_ I prefer the second option, even though it means the values are different between KafkaConnector and KMM2 while the keys are the same. WDYT @ppatierno ?

Reply:
Not sure, I _think_ I prefer the first option but I have not a strong opinion tbh. Both of them are valid. The doubt on the second option is around the fact that it could be confusing for users to see the same annotation names but getting different values (bool vs string) even if in two different resources. Anyway, I am fine with both, in the end it's a matter of documenting the difference.